### ISSUE TITLE
Opening /proc/self/fd/ causes abort()

### ISSUE THREAD

**Issue by gkv311** [emscripten-core/emscripten#10774]
Running the following simple program causes unexpected behavior - application is aborted via abort() instead of reporting file open error or actually opening the file. I haven't found in documentation that emulated file system might misbehave like this.

I don't care about simulating particular `/proc` feature (listing opened file descriptors), but this is certainly abnormal crashing application. Just tried to debug a random crash on opening `/proc/%d/status`, which unlike `/proc/self/fd/` does not exist, but still crashes somehow... (probably with the same reason).

```
#include <cerrno>
#include <dirent.h>
#include <inttypes.h>
#include <stdio.h>
int main()
{
  FILE* aFile4 = fopen ("/proc/self/fd/", "rb");
  printf ("/proc/self/fd/ FILE: %" PRIuPTR "\t[errno: %d]\n", (uintptr_t )aFile4, aFile4 == NULL ? errno : 0);
  DIR* aDir4 = opendir ("/proc/self/fd/");
  printf ("/proc/self/fd/ DIR:  %" PRIuPTR "\t[errno: %d]\n", (uintptr_t )aDir4, aDir4 == NULL ? errno : 0);
}
```

```
Output:
sample.html:1 Uncaught (in promise) abort(TypeError: Cannot read property 'parent' of undefined) at Error
    at jsStackTrace (http://localhost:8080/sample.js:1808:17)
    at stackTrace (http://localhost:8080/sample.js:1825:16)
    at abort (http://localhost:8080/-sample.js:1575:44)
    at ___syscall5 (http://localhost:8080/sample.js:5284:69)
    at _fopen (wasm-function[12960]:0x9a9f46)
    at _main (wasm-function[961]:0x145276)
    at Object.Module._main (http://localhost:8080/sample.js:12351:33)
    at callMain (http://localhost:8080/sample.js:13120:30)
    at doRun (http://localhost:8080/sample.js:13179:23)
    at run (http://localhost:8080/sample.js:13194:5)
```

Some related discussions, which I don't get:
https://groups.google.com/forum/m/#!msg/emscripten-discuss/S9v70ugb8CM/RNfa7QxVAgAJ

> I don't remember the details, but the code in library_fs.js has this comment:
> 
> // create /proc/self/fd which allows /proc/self/fd/6 => readlink gives the name of the stream for fd 6 (see test_unistd_ttyname)
> 
> We do create it in a special way. Perhaps that causes only accessing files under it to work, but reading it directly fails. Not sure offhand how easy that is to fix.
>

**Comment by dalcde** [comment]
A slightly smaller example:
```c
include <dirent.h>
#include <sys/types.h>

int main() {
  DIR *dp = opendir("/proc/self/fd");
}
```

**Comment by sbc100** [comment]
I imagine the fix should fairly straight forward.   One problem is that is looks like you are running a fairly old version of emscripten (`__syscall5` style syscalls have not existed in JS for a long time).

**Comment by dalcde** [comment]
I can reproduce this with emcc 2.0.10. Error log:
```
TypeError: Cannot read property 'parent' of undefined
exception thrown: RuntimeError: abort(TypeError: Cannot read property 'parent' of undefined) at Error
    at jsStackTrace (/tmp/a.out.js:1632:19)
    at stackTrace (/tmp/a.out.js:1649:16)
    at abort (/tmp/a.out.js:1378:44)
    at ___sys_open (/tmp/a.out.js:4037:69)
    at <anonymous>:wasm-function[6]:0x252
    at <anonymous>:wasm-function[7]:0x274
    at <anonymous>:wasm-function[4]:0x1e2
    at main (<anonymous>:wasm-function[5]:0x204)
    at /tmp/a.out.js:1435:22
    at callMain (/tmp/a.out.js:4396:15),RuntimeError: abort(TypeError: Cannot read property 'parent' of undefined) at Error
    at jsStackTrace (/tmp/a.out.js:1632:19)
    at stackTrace (/tmp/a.out.js:1649:16)
    at abort (/tmp/a.out.js:1378:44)
    at ___sys_open (/tmp/a.out.js:4037:69)
    at <anonymous>:wasm-function[6]:0x252
    at <anonymous>:wasm-function[7]:0x274
    at <anonymous>:wasm-function[4]:0x1e2
    at main (<anonymous>:wasm-function[5]:0x204)
    at /tmp/a.out.js:1435:22
    at callMain (/tmp/a.out.js:4396:15)
    at abort (/tmp/a.out.js:1384:11)
    at ___sys_open (/tmp/a.out.js:4037:69)
    at <anonymous>:wasm-function[6]:0x252
    at <anonymous>:wasm-function[7]:0x274
    at <anonymous>:wasm-function[4]:0x1e2
    at main (<anonymous>:wasm-function[5]:0x204)
    at /tmp/a.out.js:1435:22
    at callMain (/tmp/a.out.js:4396:15)
    at doRun (/tmp/a.out.js:4460:23)
    at run (/tmp/a.out.js:4475:5)
```

**Comment by sbc100** [comment]
Fix is in #13168
